File: cleancode/tree.go
package main

import "fmt"
import "strings"

func main() {
	drawTree(5, true)
	fmt.Println()
	drawTree(10, false)
}

func drawTree(height int, tip bool) {
	beforeCenter := height - 1
	spacesBeforeCenter := strings.Repeat(" ", beforeCenter);
	if tip { fmt.Println(spacesBeforeCenter + "*") }
	
	for i := 0; i < height; i++ {
		fmt.Println(spacesBeforeCenter[:beforeCenter-i] + strings.Repeat("X", 2 * i + 1))
	}
	fmt.Println(spacesBeforeCenter + "X")
}

File: cleancode/tree.java
class Tree {
	
	static void drawTree(int height, boolean tip) {
		int beforeCenter = height - 1;
		StringBuilder sb = new StringBuilder();
		if (tip) {
			sb.append(new String(new char[beforeCenter]).replace("\0", " "));
			sb.append(String.format("*%n"));
		}
		for (int i = 0; i < height; i++) {
			sb.append(new String(new char[beforeCenter - i]).replace("\0", " "));
			sb.append(new String(new char[2 * i + 1]).replace("\0", "X"));
			sb.append(String.format("%n"));
		}
		sb.append(new String(new char[beforeCenter]).replace("\0", " "));
		sb.append(String.format("X%n"));
		System.out.println(sb.toString());
	}
	
	public static void main(String[] args) {
		drawTree(5, true);
		drawTree(10, false);
	}
}